<launch>
  <node
    name="console"
    pkg="rqt_console"
    type="rqt_console" />
  <arg
    name="model" />
  <arg
    name="gui"
    default="True" />
  <param
    name="robot_description"
    textfile="$(find koko_description_threedof_kin)/robots/koko_description_threedof_kin.urdf" />
  <param
    name="/DOF/robot_dyn_description"
    textfile="$(find koko_description_threedof_dyn)/robots/koko_description_threedof_dyn.urdf" />
  <param 
    name="use_gui"
    value="$(arg gui)" />
  <rosparam file="$(find koko_bringup)/config/control_3dof.yaml" command="load"/>
  <node
    name="robot_state_publisher"
    pkg="robot_state_publisher"
    type="state_publisher" />
  <remap 
    from="/joint_state_tracker" 
    to="/DOF/joint_state_tracker" />
  <node
    name="rviz"
    pkg="rviz"
    type="rviz"
    args="-d $(find fivedof_tracker_kin_v5)/urdf.rviz" />
  <node
    name="ik_vive"
    pkg="koko_vive"
    type="ik_vive" />
  <node
    name="koko_controller_manager"
    pkg="koko_control"
    type="koko_controller_manager"
    ns="DOF" /> 
  <node
    name="joint_state_relay"
    pkg="topic_tools"
    type="relay"
    args="/DOF/joint_states /joint_states" />
  <node
    name="pose_target"
    pkg="koko_kinematics"
    type="pose_target" />
  <node
    name="visual_pos_controller_four"
    pkg="fourDOF_tracker_kin_v2"
    type="visual_controller_four" />
  <include
    file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" />
  <!--node
    name="calibration_writter"
    pkg="calibration"
    type="Writer" /-->
</launch>
